llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6743,8 +6743,14 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
     TableSize =
         (MaxCaseVal->getValue() - MinCaseVal->getValue()).getLimitedValue() + 1;
 
+  // If the default destination is unreachable, or if the lookup table covers
+  // all values of the conditional variable, branch directly to the lookup table
+  // BB. Otherwise, check that the condition is within the case range.
+  bool DefaultIsReachable = !SI->defaultDestUndefined();
+
   bool TableHasHoles = (NumResults < TableSize);
-  bool NeedMask = (TableHasHoles && !HasDefaultResults);
+  bool HolesHaveUndefinedResults = (TableHasHoles && !HasDefaultResults);
```

Comment:
This variable name is confusing to me. Could you explain it further and add comments?

Reply:
I've pushed some new comments explaining this in more detail.

Reply:
Maybe you could change it to `AllHolesAreUndefined`, `AllHolesArePoison` or `AllHolesAreUnreachable`.